src/encoder.rs

```diff
@@ -717,6 +717,34 @@ impl<T: Pixel> CodedFrameData<T> {
       imp_b_segments: Default::default(),
     }
   }
+
+  // Assumes that we have already computed activity scales and distortion scales
+  pub fn compute_spatiotemporal_scores(&mut self) -> u32 {
+    let mut scores = self
+      .distortion_scales
+      .iter()
+      .zip(self.activity_scales.iter())
+      .map(|(&d, &a)| d * a)
+      .collect::<Box<_>>();
+
+    let inv_mean = DistortionScale::inv_mean(&scores);
+
+    for score in scores.iter_mut() {
+      *score = *score * inv_mean;
```

Comment:
Could use `score *= inv_mean` (and same for `scale` below), I actually thought there was a clippy lint for this.

Reply:
Right, I'll need to include another `impl` or possibly `Derive` it.

Reply:
Ah, I see because it's adding DistortionScales. That makes sense. If you don't want to do that as part of this changeset then, that's fine.